src/styles/createTypography.d.ts

```diff
@@ -36,7 +36,11 @@ export interface TypographyStyle {
 
 export type Typography = { [type in Style]: TypographyStyle } & FontStyle;
 
-export type TypographyOptions = Partial<FontStyle> & Partial<Typography>;
+export type DeepPartial<T> = {
+  [P in keyof T]?: DeepPartial<T[P]>;
+};
```

Comment:
This should probably go in `index.d.ts` along with `Diff` and `Omit`... we will need it to fix #9333 too.